### PR TITLE
fix: Handle unhandled error in `startResumableUpload_`

### LINKED ISSUE #2494: Unhandled Error in startResumableUpload_ Function
Cannot handle the error by `try...catch` nor `writable.on('error')`

#### Environment details

  - OS: All
  - Node.js version: All
  - npm version: All
  - `@google-cloud/storage` version: 7.11.2

#### Steps to reproduce

  1.  partial upload without url provided
  2. run `file.createWriteStream`
  

```javascript
const opitons = {
  offset: 1,
  isPartialUpload: true,
};
let writable;
try {
   writable = file.createWriteStream(opitons);
} catch (error) {
  // cannot catch the error
  return;
}

writable.on('error', (err: RangeError) => {
   // cannot catch the error
});

writable.write('data');
```

### PR DESCRIPTION
**Pull request by cuyl**
* Added error handling using `stream.destroy`

- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #2494

### CODE REVIEW

File: src/file.ts [hunk 1]
```diff
@@ -4168,7 +4167,17 @@ class File extends ServiceObject<File, FileMetadata> {
...
-    });
+    };
+
+    let uploadStream: resumableUpload.Upload;
+
+    try {
+      uploadStream = resumableUpload.upload(cfg);
+    } catch (error) {
+      dup.destroy(error as Error);
+      this.storage.retryOptions.autoRetry = false;
```
Comment (ddelgrosso1) [inline src/file.ts:4178]: There is no need to set this here.

Comment (cuyl) [reply src/file.ts:4178]: Seems this is a unrecoverable error.

https://github.com/googleapis/nodejs-storage/blob/18eef67892af6c7311a12d4c22579e240149eb02/src/resumable-upload.ts#L350-L354

Comment (ddelgrosso1) [reply src/file.ts:4178]: Retries for resumable uploads do not happen at this layer. Turning this to false here is shutting off retries for other operations as well.

Comment (cuyl) [reply src/file.ts]: OK, NP

**Comment by ddelgrosso1** [comment]
Thank you for the contribution @cuyl, we appreciate it.

**Review by ddelgrosso1 — APPROVED** [review]

**Pull request merged**